nautobot/dcim/views.py

```diff
@@ -3150,3 +3177,67 @@ def get_extra_context(self, request, instance):
             members_table.columns.show("device_redundancy_group_priority")
             context["members_table"] = members_table
         return context
+
+
+class InterfaceRedundancyGroupUIViewSet(NautobotUIViewSet):
+    """ViewSet for the InterfaceRedundancyGroup model."""
+
+    bulk_update_form_class = forms.InterfaceRedundancyGroupBulkEditForm
+    filterset_class = filters.InterfaceRedundancyGroupFilterSet
+    filterset_form_class = forms.InterfaceRedundancyGroupFilterForm
+    form_class = forms.InterfaceRedundancyGroupForm
+    queryset = InterfaceRedundancyGroup.objects.select_related("status")
+    queryset = queryset.prefetch_related("interfaces")
+    queryset = queryset.annotate(
+        interface_count=count_related(Interface, "redundancy_groups"),
+    )
+    serializer_class = serializers.InterfaceRedundancyGroupSerializer
+    table_class = tables.InterfaceRedundancyGroupTable
+    lookup_field = "pk"
+
+    def _process_bulk_create_form(self, form):
+        """Bulk creating (CSV import) is not supported."""
+        raise NotImplementedError()
+
+    def get_extra_context(self, request, instance):
+        """Return additional panels for display."""
+        context = super().get_extra_context(request, instance)
+        if instance and self.action == "retrieve":
+            interface_table = self._get_interface_redundancy_groups_table(request, instance)
+            context["interface_table"] = interface_table
+        return context
+
+    def _get_interface_redundancy_groups_table(self, request, instance):
+        """Return a table of assigned Interfaces."""
+        queryset = instance.interfaceredundancygroupassociation_set.restrict(request.user)
+        queryset = queryset.prefetch_related("interface")
+        queryset = queryset.order_by("priority")
+        column_sequence = (
+            "interface__device",
+            "interface",
+            "priority",
+            "interface__status",
+            "interface__enabled",
+            "interface__ip_addresses",
+            "interface__type",
+            "interface__description",
+            "interface__label",
+        )
+        table = tables.InterfaceRedundancyGroupAssociationTable(
+            data=queryset,
+            sequence=column_sequence,
+            orderable=False,
+        )
+        for column_name in column_sequence:
+            table.columns.show(column_name)
+        return table
+
+
+class InterfaceRedundancyGroupAssociationCreateView(generic.ObjectEditView):
```

Comment:
I wonder if we can do something like this here 
```suggestion
class InterfaceRedundancyGroupAssociationUIViewSet(mixins.ObjectEditViewMixin, mixins.ObjectDestroyViewMixin):
```

Reply:
fixed